src/mempool/own_lane.rs

```diff
@@ -95,6 +100,11 @@ impl super::Mempool {
             .get_pending_entries_in_lane(&self.own_lane_id(), last_cut)?;
 
         for (entry_metadata, dp_hash) in entries {
+            if let Some(ref dp_hash_hint) = dp_hash_hint {
+                if &dp_hash != dp_hash_hint {
+                    break;
```

Comment:
maybe a comment on why we break there ? (or either update the name of gp_hash_hint to explain the reason ^^ but by reading the PR, I don't get it why we break)

Reply:
Added a comment & renamed - the idea is that if `only_dp_with_hash` is Some, then we only disseminate that one, if None, we act as usual. It's an optimisation so that create-new-dp only disseminates the DP we just created